server/api/resolvers/requestGroupResolvers.ts

```diff
@@ -0,0 +1,139 @@
+        return nextRequestEmbeddingForRequestType(requestType, session)
+    }))
+
+    if (nextRequestsForAllRequestTypes === null) {
+        return null
+    }
+
+    const sortedNextRequests = nextRequestsForAllRequestTypes
+        .sort((requestEmbedding1, requestEmbedding2) => {
+            if (requestEmbedding1 === null) {
+                return 1
+            }
+
+            if (requestEmbedding2 === null) {
+                return -1
+            }
+            
+            return requestEmbedding1.createdAt.getMilliseconds() - requestEmbedding2.createdAt.getMilliseconds()
+        })
+    
+    if (sortedNextRequests.length === 0) {
+        return null
+    }
+
+    return sortedNextRequests[0]
+}
+
+const requestGroupQueryResolvers = {
+    requestGroup: async (_, { _id }, ___): Promise<RequestGroupInterface> => {
+        return RequestGroup.findById(_id).exec()
+    },
+    requestGroups: async (_, __, ___): Promise<Array<RequestGroupInterface>> => {
+        return RequestGroup.find().exec()
+    },
+    requestGroupsFilter: async (_, { filter, options }, ___): Promise<Array<RequestGroupInterface>> => {
+        return RequestGroup.find().exec()
+    },
+}
+
+const requestGroupMutationResolvers = {
+    createRequestGroup: async (_, { requestGroup }, { authenticateUser }): Promise<RequestGroupInterface> => {
+        return authenticateUser().then(async () => { 
+            const newRequestGroup = new RequestType({...requestGroup})
+            const createdRequestGroup = await newRequestGroup.save()
+
+            return createdRequestGroup
+        })
+    },
+    updateRequestGroup: async (_, { requestGroup }, { authenticateUser }): Promise<RequestGroupInterface> => {
+        return authenticateUser().then(async () => {
+            const oldRequestGroup = await RequestGroup.findById(requestGroup._id)
+            const modifiedRequestGroup = new RequestType({...oldRequestGroup, ...requestGroup})
+
+            return modifiedRequestGroup.save()
+        })
+    },
+    deleteRequestGroup: async (_, { _id }, { authenticateUser }): Promise<RequestGroupInterface> => {
+        return authenticateUser().then(async () => {
+            const requestGroup = await RequestGroup.findById(_id)
+            requestGroup.deletedAt = new Date()
+            return requestGroup.save()
+        })
+    },
+}
+
+const requestGroupResolvers = {
+    requestTypes: async (parent, __, ___): Promise<Array<RequestTypeInterface>> => {
+        return parent.requestTypes.map((requestTypeEmbedding) => {
+            return RequestType.findById(requestTypeEmbedding._id)
+        })
+    },
+    deleted: (parent, __, ___): boolean => {
+        return parent.deletedAt !== undefined
+    },
+    countOpenRequests: async (parent, __, ___): Promise<number> => {
+        const countOpenRequestsPerType: Array<number> = await Promise.all(parent.requestTypes.map(async (requestTypeEmbedding) => {
```

Comment:
why do some resolvers require authenticateUser() but others dont?

Reply:
Depends on whether they are accessing Client information. We have restricted access to that portion of the API to authenticated users only

Reply:
yup, authenticateUser() is used to restrict usage of mutation endpoints and endpoints that expose client info